test/functional/ex_cmds/swapfile_preserve_recover_spec.lua

```diff
@@ -373,3 +377,90 @@ describe('swapfile detection', function()
     feed('e')
   end)
 end)
+
+describe('quitting swapfile dialog on startup stops TUI properly', function()
+  local swapdir = luv.cwd()..'/Xtest_swapquit_dir'
+  local testfile = 'Xtest_swapquit_file1'
+  local otherfile = 'Xtest_swapquit_file2'
+  -- Put swapdir at the start of the 'directory' list. #1836
+  -- Note: `set swapfile` *must* go after `set directory`: otherwise it may
+  -- attempt to create a swapfile in different directory.
+  local init_dir = [[set directory^=]]..swapdir:gsub([[\]], [[\\]])..[[//]]
+  local init_set = [[set swapfile fileformat=unix nomodified undolevels=-1 nohidden]]
+
+  before_each(function()
```

Comment:
This setup code looks redundant with the `describe('swapfile detection'` setup code. Can we combine these into one describe() ?

Reply:
This `before_each()` does more things than that one, so another `describe()` block is needed anyway.